### PR TITLE
when update gateway info, we should append old to new deploy

### PR DESCRIPTION
**Pull request by wangyd1988**
#### What type of this PR
   
meet user  defined annotation requirements

### CODE REVIEW

File: pkg/controller/vpc_nat_gateway.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func (c *Controller) handleAddOrUpdateVpcNatGw(key string) error {
 
 	// check or create deployment
 	needToCreate := false
-	_, err = c.config.KubeClient.AppsV1().Deployments(c.config.PodNamespace).
+	oldDp, err := c.config.KubeClient.AppsV1().Deployments(c.config.PodNamespace).
```
Comment (oilbeater) [inline pkg/controller/vpc_nat_gateway.go:249]: We should make a deepcopy of the oldDp, other wise the `newPodAnnotations[key] = value` operations below will affect the cached value.

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/vpc_nat_gateway.go (hunk 1 above)
Comment (wangyd1988) [reply pkg/controller/vpc_nat_gateway.go:249]: it's only get  old  cached value from oldDp, how  affect?

Comment (oilbeater) [reply pkg/controller/vpc_nat_gateway.go:249]: https://go.dev/play/p/63SXnSpqSNp

Comment (wangyd1988) [reply pkg/controller/vpc_nat_gateway.go:249]: > https://go.dev/play/p/63SXnSpqSNp

good ，i see

Comment (wangyd1988) [reply pkg/controller/vpc_nat_gateway.go:249]: But I find  that this client doesn't have caching
image
image

Comment (oilbeater) [reply pkg/controller/vpc_nat_gateway.go:249]: you're right, i didn't notice it

**Review by oilbeater — APPROVED** [review]

**Pull request merged**